samples/runtime/qpic-simulator/host/Program.qs

```diff
@@ -48,10 +49,7 @@ namespace Microsoft.Quantum.Samples {
         // has no effect in other simulators)
         Barrier();
         CNOT(b, a);
-        if (M(a) == One) {
-            S(b);
-            X(a);
-        }
+        Extensions.ApplyIfOne(MResetZ(a), (S, b));
```

Comment:
The APIs at https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.simulation.quantumprocessor.extensions are not documented; could you add a small comment here explaining what's happening?